### ISSUE TITLE
Question about the result of open images dataset

### ISSUE THREAD

**Issue by AhLok-Chow** [deepcs233/Visual-CoT#15]
Hi, thank you for your amazing work.
I'm trying to run the code and reproduce the result mentioned in the paper. I test the viscot-7b-336 on the relationship reasoning dataset(gqa, open images, and vsr) and the accuracy are 0.635, 0.599 and 0.570 respectively. I wonder why the accuracy on the open images dataset is different from that in the paper (but the results on gqa and vsr are similar), and I don't know whether my reproduction is wrong. 
Hopefully I can get your help.

**Comment by deepcs233** [comment]
Hi!

Could you provide more details about your reproduction process? Specifically, did you re-train the model using our dataset, or did you simply evaluate the pre-trained model on our benchmark? Regarding the Open Images dataset, we only used a subset for training and evaluation, which might explain some of the discrepancies you're noticing.

**Comment by AhLok-Chow** [comment]
> Hi!
> 
> Could you provide more details about your reproduction process? Specifically, did you re-train the model using our dataset, or did you simply evaluate the pre-trained model on our benchmark? Regarding the Open Images dataset, we only used a subset for training and evaluation, which might explain some of the discrepancies you're noticing.


My reproduction process:

1. Clone this repository and install package.
2. Download the Viscot-7b-336 checkpoints from HF, download the image files from HF link, then extract them.
3. Use the evaluation code to get result. (Here I fix some bugs according to the similar issues in llava)
4. Obtain the score by GPT-4o.

Code: This repository
Data: From HF link
Checkpoints: From HF link
API: GPT-4o

I only do steps above on relationship reasoning part. Did I get the images wrong?

**Comment by deepcs233** [comment]
Hi!

We use GPT-3.5 for faster and cheaper evaluation, can you try again?

**Comment by AhLok-Chow** [comment]
I try it again and the accuracy is still 0.599. It seems strange.

**Comment by deepcs233** [comment]
Hi!

It seems odd that both APIs are receiving the same score. Could you double-check the prompt or verify if the correct API is being used?

**Comment by AhLok-Chow** [comment]
Yeah, I double check the prompt and make sure it is the correct API. I get result as follow to avoid some Internet problem:

client = OpenAI(
    api_key = "",  
)
response = client.chat.completions.create(
    model="gpt-3.5-turbo-1106",
    # response_format={"type": "json_object"},
    messages=[
        {"role": "system", "content": BASE_PROMPT},
        {"role": "user", "content": content},
    ],
)
response = response.choices[0].message.content
print(response)
return response

And the code below is the original code:

data = {}
data['model']= "gpt-3.5-turbo-1106"
data['messages'] = [{"role":"system","content": BASE_PROMPT}, {"role": "user", "content":content}]
for key in extra_args:
     data[key] = extra_args[key]
headers['Authorization'] = API_KEY
r = requests.post('https://api.openai.com/v1/chat/completions', headers=headers, json=data, timeout=60)
response = r.json()

I think it can get the same result because I find that the score given by GPT in scores.json file is correct.
